src/core/typing.ts

```diff
@@ -214,8 +200,7 @@ export class DefaultTyping
 
     // Start typing and emit typingStarted event
     this._startTypingTimer();
-    const channel = await this.channel;
-    return channel.presence.enterClient(this._clientId).then();
+    return this._channel.presence.enterClient(this._clientId).then();
```

Comment:
Not related to your PR, but I think we can remove the .then() at the end?

Reply:
A wild `then` appeared!